cli.js

```diff
@@ -1,124 +1,146 @@
+
+        To get help with a command
+            lume COMMAND --help
+
+    USAGE:
+        lume [COMMAND] [OPTIONS]
+
+    COMMANDS:
+        build      Build the site. It's the default command
+        init       Creates a _config.js file
+        run        Run an site script
+        upgrade    Upgrade 🔥lume to the latest version
+        update     Update the version of the lume modules imported in a _config.js file
+
+
+    OPTIONS:
+        -h, --help              print usage information
+        -v, --version  Prints version information
+            --root <dir>        the root that lume should work in   Default: ./
+            --src  <dir>        the source directory for your site  Default: ./
+            --dest  <dir>       the build destination.              Default: _site
+            --config <file>     specify the lume config file.       Default: _config.js
+            --location <domain> set the domain for your site.       Default: http://localhost
+            --dev               enable dev mode (view draft pages)
+            
+            --serve             start a live-reloading web server
+            --port <port>       the port the server is on           Default: 3000
+`;
+
 
 if (import.meta.main) {
-  cli(Deno.args);
+  await cli(Deno.args);
 }
 
 export default async function cli(args) {
-  const version = "v0.14.0";
-  let stop = false;
-  const options = parse(args, {
-    boolean: ["serve", "init", "version", "dev", "help", "upgrade", "update"],
-    string: ["run", "port", "src", "dest", "location", "root", "config"],
-    alias: {
-      help: "h",
-      version: "v",
-    },
-    ["--"]: true,
-    unknown(option) {
-      if (option.startsWith("-")) {
-        console.log(`Unknown option: ${option}`);
-        stop = true;
-      }
-    },
-    default: {
-      root: Deno.cwd(),
-      config: "_config.js",
-    },
+  // the rest of the option parsing is handled within each command
+  let options = parse(args, {
+    boolean: ["help", "version"],
+    alias: {help: "h", version: "v"},
   });
 
-  const command = options._[0] || "build";
-
-  if (stop) {
-    console.log(`Run ${brightGreen("lume --help")} for usage information`);
-    console.log("");
+  // lume --version
+  if (options.version) {
+    console.log(`🔥lume ${version}`);
     return;
   }
 
   if (options._.length > 1) {
-    console.log(`Too much arguments: ${options._.join(", ")}`);
+    console.log(`Too many arguments: ${options._.join(", ")}`);
     console.log(`Run ${brightGreen("lume --help")} for usage information`);
     console.log("");
-    return;
+    Deno.exit(1);
   }
 
-  // lume --help
+  // _ contains the non-option arguments
+  const command = options._[0]?.toLowerCase();
+
+  // lume [COMMAND] --help
   if (options.help) {
-    return help(version);
+    let usage;
+    if (command) {
+      const commandModule = await import(`./cli/${command}.js`).catch(_ => {
+      }) //ignore import errors here
```

Comment:
Why ignore the errors here?

Reply:
Maybe, if we import the commands as `import * as build from "./cli/build.js"`, we don't need to import again here, just run `build.USAGE` (that i'd rename to `help`, for coherence with the option name).

Reply:
I think the only time an error would occur is if the command doesn't exist and in that case we should tell them.

Reply:
I'm ignoring it here because it would mean the command doesn't exist, which is a scenario handled at the end of the command checks. 

@oscarotero I'll play around with other ways to import, to try to avoid this dynamic import